#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_zabbix_suckerd.c by Michael Ablassmeier <abi@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: re-use socket on startup 

@DPATCH@
--- server.c	2006/06/02 14:04:04	1.75
+++ zabbix-1.1/src/zabbix_server/server.c	2006/06/24 12:11:43	1.76
@@ -408,6 +408,7 @@
 {
 	int	sockfd;
 	struct	sockaddr_in      serv_addr;
+	int	on;
 /*	struct linger ling;*/
 
 	if ( (sockfd = socket(AF_INET, SOCK_STREAM, 0)) == -1)
@@ -416,6 +417,15 @@
 		exit(1);
 	}
 
+	/* Enable address reuse */
+	/* This is to immediately use the address even if it is in TIME_WAIT state */
+	/* http://www-128.ibm.com/developerworks/linux/library/l-sockpit/index.html */
+	on = 1;
+	if( -1 == setsockopt( sockfd, SOL_SOCKET, SO_REUSEADDR, &on, sizeof(on) ))
+	{
+		zabbix_log(LOG_LEVEL_WARNING, "Cannot setsockopt SO_REUSEADDR [%s]", strerror(errno));
+	}
+
 /*	if(CONFIG_NOTIMEWAIT == 1)
 	{
 		ling.l_onoff=1;
